### PR TITLE
updated COSE envelope for the certification chain changes

### PR DESCRIPTION
**Pull request by patrickzheng200**
Signed-off-by: Patrick Zheng [patrickzheng@microsoft.com](mailto:patrickzheng@microsoft.com)

### CODE REVIEW

File: signature/cose/envelope.go [hunk 1]
```diff
@@ -382,22 +421,23 @@ func generateProtectedHeaders(req *signature.SignRequest, protected cose.Protect
...
 	// signing agent
 	unprotected[headerLabelSigningAgent] = req.SigningAgent
 
 	// certChain
-	certs, err := req.Signer.CertificateChain()
-	if err != nil {
-		return err
+	var certs []*x509.Certificate
+	switch s := signer.(type) {
+	case *remoteSigner:
```
Comment (binbin-li) [inline signature/cose/envelope.go:431]: nit: `case *remoteSigner, *localSigner:`

**Review by binbin-li — APPROVED** [review]
one minor comment, lgtm overall

File: signature/cose/envelope.go (hunk 1 above)
Comment (patrickzheng200) [reply signature/cose/envelope.go:431]: @binbin-li I actually have tried that but got the following error: s.certs undefined (type "github.com/veraison/go-cose".Signer has no field or method certs).
variable s becomes cose.Signer type, and it does not have certs field defined, so looks like I have to separate them into 2 cases.

Comment (binbin-li) [reply signature/cose/envelope.go:431]: gotcha, good to know that.

**Pull request merged**